designs/senya/src/bottom.mjs

```diff
@@ -0,0 +1,158 @@
+  sa,
+  points,
+  Path,
+  paths,
+  Snippet,
+  snippets,
+  options,
+  measurements,
+  complete,
+  paperless,
+  macro,
+  part,
+  log,
+}) {
+  let topCircumverence = measurements.underbust
+  let fullLength =
+    measurements.hpsToWaistBack -
+    waistlineHeight(
+      measurements.waistToUnderbust,
+      measurements.hpsToWaistBack,
+      options.lowerWaistline
+    ) +
+    measurements.waistToHips
+  let angle = 180 * options.circleRatio
+  let radiusMidriffTop = topCircumverence / utils.deg2rad(angle) / 2
+
+  if (angle > 90 && sa) {
+    const maxAngle = utils.rad2deg(Math.atan(radiusMidriffTop / sa))
+    if (angle > 90 + maxAngle) angle = 90 + maxAngle
+  }
+
+  const radiusHem = radiusMidriffTop + fullLength * options.lengthBonus
+
+  const ids = macro('ringsector', {
+    angle: angle,
+    insideRadius: radiusMidriffTop,
+    outsideRadius: radiusHem,
+    rotate: true,
+  })
+  const pathId = ids.paths.path
+  paths.seam = paths[pathId].clone().addClass('fabric')
+  paths[pathId].hide()
+
+  for (const [shortId, uid] of Object.entries(ids.points)) {
+    points[shortId] = points[uid].copy()
+    // Some points are rotated, we need those too
+    if (points[uid + 'Rotated']) points[shortId + 'Rotated'] = points[uid + 'Rotated'].copy()
+  }
+
+  points.gridAnchor = points.in2Flipped.clone()
+
+  // Complete pattern?
+  if (complete) {
+    macro('cutonfold', {
+      from: points.in2Flipped,
+      to: points.ex2Flipped,
+      grainline: true,
+    })
+    points.logo = points.in2FlippedRotated.shiftFractionTowards(points.ex2FlippedRotated, -0.1)
+    snippets.logo = new Snippet('logo', points.logo)
+
+    points.title = points.in2FlippedRotated.shiftFractionTowards(points.ex2FlippedRotated, -0.5)
+    macro('title', { at: points.title, nr: 3, title: 'bottom' })
+
+    macro('sprinkle', {
+      snippet: 'notch',
+      on: ['in1Rotated', 'gridAnchor'],
+    })
+
+    if (sa) {
+      paths.hemBase = new Path()
+        .move(points.ex1Rotated)
+        .curve(points.ex1CFlippedRotated, points.ex2CFlippedRotated, points.ex2FlippedRotated)
+        .curve(points.ex1CFlipped, points.ex2CFlipped, points.ex2Flipped)
+        .offset(fullLength * options.lengthBonus * options.hemWidth * -1)
+      paths.saBase = new Path()
+        .move(points.in2Flipped)
+        .curve(points.in2CFlipped, points.in1CFlipped, points.in2FlippedRotated)
+        .curve(points.in2CFlippedRotated, points.in1CFlippedRotated, points.in1Rotated)
+      paths.saBase = paths.saBase.line(points.ex1Rotated)
+      paths.saBase = paths.saBase.offset(sa * -1)
+
+      paths.hemBase.hide()
+      paths.saBase.hide()
+
+      paths.sa = new Path()
+        .move(points.in2Flipped)
+        .line(paths.saBase.start())
+        .join(paths.saBase)
+        .line(paths.hemBase.start())
+        .join(paths.hemBase)
+        .line(points.ex2Flipped)
+        .attr('class', 'fabric sa')
+    }
+  }
+
+  // Paperless?
+  if (paperless) {
+    macro('vd', {
```

Comment:
All macros need to get an `id`. If not, you will only be able to use the same macro once per part. So give all these dimensions an id, as you've already done for the ones you want to be able to remove later.

(as it happens, making sure they can be removed is why we require an id now)

Reply:
Huh, cool, now all of them show up, previously (before v3 I guess) it was just the first n for some n.

In bad news while doing this I realized that when the circle ratio differs from 50% then the printed distances seem... useless? Unless it's just me being unable to understand how I am supposed to be using them. I checked with Sandy, but it does the exact same thing, so maybe I'm just wrong.

Reply:
I think there might be a bug with Sandy's dimensions for non-50% circles.